src/WeightList.php

```diff
@@ -32,7 +32,7 @@ private function getValueByPosition($position)
             }
         }
 
-        return $value;
+        throw new \OutOfBoundsException('Index of ' . $position . ' is out of bounds for the array.');
```

Comment:
$position is not index of array, if points to waighted element of priority list and may be any value from 0 to sum of all elemet's wegiths. also it's better to move check for empty array before iterator

Reply:
Ok, I check the array count is zero before iterator.